src/kbmod/result_list.py

```diff
@@ -382,13 +439,41 @@ def filter_on_stats(self, lh_threshold=-1.0, min_valid_indices=-1):
             The minimum number of valid indices a row needs to pass the filtering.
             Use -1 to ignore this field.
         """
-        tmp_results = []
-        for x in self.results:
+        indices_to_keep = []
+        for i, x in enumerate(self.results):
             keep = min_valid_indices == -1 or len(x.valid_indices) >= min_valid_indices
             keep = keep and (lh_threshold < 0.0 or x.final_likelihood >= lh_threshold)
             if keep:
-                tmp_results.append(x)
-        self.results = tmp_results
+                indices_to_keep.append(i)
+        self.filter_results(indices_to_keep, "filter_on_stats")
```

Comment:
Could the user provide their own filter label?

If, say, they do something like `resultList.filter_on_stats(easy_stats).filter_on_more(more_stats).filter_on_stats(expensive_stats)`, it would be tough to distinguish what gets filtered because of the first call to `filter_on_stats` vs the second call.

Maybe pre-pend some filter counter to the label? Or maybe no one cares?

Reply:
You are ahead of the game. In the next PR the filter class will dynamically generate the labels based on both the filter and the parameters.